File: ts/project/.sdk/tm/rb/test/netsim_test.rb
# ProjectName SDK netsim test
#
# Network-behaviour simulation over the offline mock transport. The test
# feature accepts an optional "net" config so unit tests can exercise slow,
# failing and offline conditions without a live server. These checks drive
# the transport through direct(), which needs no entity, so they run for
# every generated SDK regardless of its API shape.

require "minitest/autorun"
require_relative "../ProjectName_sdk"

class NetsimTest < Minitest::Test
  def test_offline_simulation_fails_the_request
    sdk = ProjectNameSDK.test({ "net" => { "offline" => true } }, nil)
    res = sdk.direct({ "path" => "/ping" })
    assert_equal false, res["ok"], "offline network must fail the call"
    refute_nil res["err"]
  end

  def test_fail_status_simulation_surfaces_the_error_status
    sdk = ProjectNameSDK.test({ "net" => { "failTimes" => 1, "failStatus" => 503 } }, nil)
    res = sdk.direct({ "path" => "/ping" })
    assert_equal false, res["ok"]
    assert_equal 503, res["status"], "simulated failure status is surfaced"
  end

  def test_error_times_simulation_yields_a_connection_error
    sdk = ProjectNameSDK.test({ "net" => { "errorTimes" => 1 } }, nil)
    res = sdk.direct({ "path" => "/ping" })
    assert_equal false, res["ok"]
    assert_match(/connection error/i, res["err"].to_s)
  end

  def test_latency_simulation_delays_the_request
    delay = 60
    sdk = ProjectNameSDK.test({ "net" => { "latency" => delay } }, nil)
    start = Time.now.to_f
    sdk.direct({ "path" => "/ping" })
    elapsed = ((Time.now.to_f - start) * 1000).to_i
    # Generous lower bound to stay robust on slow CI.
    assert elapsed >= delay - 25, "expected >= #{delay - 25}ms latency, got #{elapsed}ms"
  end

  def test_plain_test_sdk_still_works_with_no_net_simulation
    sdk = ProjectNameSDK.test(nil, nil)
    refute_nil sdk
  end
end

File: ts/project/.sdk/tm/php/test/NetsimTest.php
<?php
declare(strict_types=1);

// ProjectName SDK netsim test
//
// Network-behaviour simulation over the offline mock transport. The `test`
// feature accepts an optional `net` config so unit tests can exercise slow,
// failing and offline conditions without a live server. These checks drive
// the transport through `direct()`, which needs no entity, so they run for
// every generated SDK regardless of its API shape. Mirrors
// tm/ts/test/netsim.test.ts.

require_once __DIR__ . '/../projectname_sdk.php';
require_once __DIR__ . '/Runner.php';

use PHPUnit\Framework\TestCase;

class NetsimTest extends TestCase
{
    public function test_offline_simulation_fails_the_request(): void
    {
        $sdk = ProjectNameSDK::test(['net' => ['offline' => true]]);
        $res = $sdk->direct(['path' => '/ping']);
        $this->assertFalse($res['ok'], 'offline network must fail the call');
        $this->assertNotNull($res['err'] ?? null);
    }

    public function test_fail_status_simulation_surfaces_the_error_status(): void
    {
        $sdk = ProjectNameSDK::test(['net' => ['failTimes' => 1, 'failStatus' => 503]]);
        $res = $sdk->direct(['path' => '/ping']);
        $this->assertFalse($res['ok']);
        $this->assertSame(503, $res['status'], 'simulated failure status is surfaced');
    }

    public function test_latency_simulation_delays_the_request(): void
    {
        $delay = 60;
        $sdk = ProjectNameSDK::test(['net' => ['latency' => $delay]]);
        $start = microtime(true);
        $sdk->direct(['path' => '/ping']);
        $elapsed_ms = (microtime(true) - $start) * 1000.0;
        // Generous lower bound to stay robust on slow CI.
        $this->assertGreaterThanOrEqual($delay - 25, $elapsed_ms,
            "expected >= " . ($delay - 25) . "ms latency, got {$elapsed_ms}ms");
    }

    public function test_a_plain_test_sdk_still_works_with_no_net_simulation(): void
    {
        $sdk = ProjectNameSDK::test();
        $this->assertNotNull($sdk);
    }
}
